src/map/lua/lua_baseentity.cpp

```diff
@@ -12884,7 +12888,7 @@ uint8 CLuaBaseEntity::getModelSize()
 
 void CLuaBaseEntity::setMobFlags(uint32 flags, uint32 mobid)
 {
-    XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+    XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB);
```

Comment:
Just following up on this as well, we really should move away from these breaks in Lua Land and move towards logging output in debug.

Reply:
Ah, that got left in there when I was trying to figure out why it wasn't working the way I was expected at the time. I should have removed the break line entirely, since I need this function for the mob to change its own flag on spawn. Instead, it should just be a check in the function rather than a break.

Reply:
lil trivia:

I had actually been intending for setMobFlags and player nameflag functions and npc's entity flags related functions to all be combined (its the same stuff on diff entities, so its a lot of duplication!) but each entity type is slightly off from the others in where the bits land. so they either need bit shifting or more preferable, rework the masks. There are more bits that belong in those masks that we aren't using - far to much for one single mask field (retail is using EIGHT mask fields for all this crap!!) so its on the back burner till thats done.